src/snowflake/connector/pandas_tools.py

```diff
@@ -178,6 +183,54 @@ def write_pandas(
     else:
         columns = ",".join(list(df.columns))
 
+    if auto_create_table:
+        file_format_name = None
+        while True:
+            try:
+                file_format_name = (
+                    '"'
+                    + "".join(random.choice(string.ascii_lowercase) for _ in range(5))
+                    + '"'
+                )
+                file_format_sql = (
+                    f"CREATE FILE FORMAT {file_format_name} "
+                    f"/* Python:snowflake.connector.pandas_tools.write_pandas() */ "
+                    f"TYPE=PARQUET COMPRESSION={compression_map[compression]}"
+                )
+                logger.debug("creating file format with '{}'".format(file_format_sql))
+                cursor.execute(file_format_sql, _is_internal=True).fetchall()
+                break
+            except ProgrammingError as pe:
+                if pe.msg.endswith("already exists."):
+                    continue
+                raise
+        infer_schema_sql = f"SELECT COLUMN_NAME, TYPE FROM table(infer_schema(location=>'@\"{stage_name}\"', file_format=>'{file_format_name}'))"
+        logger.debug(f"inferring schema with {infer_schema_sql}")
```

Comment:
Keeping consistent with already existing logging messages
```suggestion
        logger.debug(f"inferring schema with '{infer_schema_sql}'")
```